pkgs/tools/admin/mcumgr/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, buildGoModule, fetchFromGitHub, testers, mcumgr }:
+
+buildGoModule rec {
+  pname = "mcumgr";
+  version = "unstable-2022-10-04";
+
+  src = fetchFromGitHub {
+    owner = "apache";
+    repo = "mynewt-mcumgr-cli";
+    rev = "5c56bd24066c780aad5836429bfa2ecc4f9a944c";
+    sha256 = "sha256-WyaDnCRyZRwOkZyzNp1ouhNY5HuAvU6U3yUjiB5m+uE=";
+  };
+
+  vendorSha256 = "sha256-EPwV47zeBqxQPtD9QeQTKB64PmMt4HYseg38cI/owwE=";
+
+  patchPhase = ''
```

Comment:
Use `postPatch` instead.

Reply:
Won't this mean that people won't be able to define their own postPatch in an overrideAttrs?

Reply:
They can.
I will enter in a more technical discussion...

Usually the default `patchPhase` patch the source code by applying the files listed in `patches`, between the `prePatch` and `postPatch` hooks.

If you write a custom `patchPhase`, then it will not be possible to apply `patches` (at least not easily).

Then you should use `postPatch` instead of a full custom `patchPhase`.